src/main/java/io/cryostat/net/web/http/api/beta/RecordingMetadataLabelsPostHandler.java

```diff
@@ -127,15 +123,12 @@ public IntermediateResponse<Metadata> handle(RequestParameters params) throws Ex
             recordingArchiveHelper.getRecordingPath(recordingName).get();
 
             Metadata updatedMetadata =
-                    recordingMetadataManager.setRecordingMetadata(recordingName, metadata).get();
-
-            notificationFactory
-                    .createBuilder()
-                    .metaCategory(RecordingMetadataManager.NOTIFICATION_CATEGORY)
-                    .metaType(HttpMimeType.JSON)
-                    .message(Map.of("recordingName", recordingName, "metadata", updatedMetadata))
-                    .build()
-                    .send();
+                    recordingMetadataManager
+                            .setRecordingMetadata(sourceTarget, recordingName, metadata)
+                            .get();
+
+            recordingMetadataManager.notifyRecordingMetadataUpdated(
```

Comment:
Are there cases where we do `setRecordingMetadata()` and don't follow it with a `notifyRecordingMetadataUpdated()`? If not, could `setRecordingMetadata()` emit the notification internally/implicitly so that callers don't need to do both?

Reply:
For cryostat-generated labels, such as the `rule=something` and `template.name=continuous`, these call `setRecordingMetadata` when a recording gets created. In this case, I don't want a notification because the user didn't specifically update the metadata themselves.

Reply:
Fair enough. I think we have a similar case elsewhere that @hareetd worked on where there's a boolean parameter that controls the notification emission? That might be better than having the notification emitted as a separate step, which is easy for code authors to just forget to add.